browser/speedreader/speedreader_tab_helper.cc

```diff
@@ -302,6 +344,36 @@ void SpeedreaderTabHelper::DidStopLoading() {
   }
 }
 
+void SpeedreaderTabHelper::DOMContentLoaded(
+    content::RenderFrameHost* render_frame_host) {
+  if (!PageWantsDistill(distill_state_))
+    return;
+
+  constexpr int kIsolatedWorldId = content::ISOLATED_WORLD_ID_CONTENT_END + 1;
+
+  constexpr const char16_t kAddShowOriginalPageLink[] =
+      uR"js(
+    (function() {
+      const link = document.getElementById('show-original-page');
+      if (!link)
+        return;
+      link.text = '$1';
+      link.addEventListener('click', (e) => {
+        window.speedreader.showOriginalPage();
+      })
+    })();
+  )js";
+
+  const auto script = base::ReplaceStringPlaceholders(
+      kAddShowOriginalPageLink,
+      brave_l10n::GetLocalizedResourceUTF16String(
+          IDR_SPEEDREADER_SHOW_ORIGINAL_PAGE_LINK),
+      nullptr);
+
+  render_frame_host->ExecuteJavaScriptInIsolatedWorld(script, base::DoNothing(),
```

Comment:
just wondering does this page have a content security policy that would prevent inline JS execution? if so i assume this kind of injected JS is exempt from the policy?

Reply:
you can make a test for this by adding an html page that contains `<head><meta http-equiv="Content-Security-Policy" content="default-src 'self'"></head>` and seeing if the link gets injected

Reply:
we have this CSP in speedreader: https://github.com/brave/brave-core/blob/05f2a4191f0dd1a50d62e7a40c9fb9506fe53f57/components/speedreader/speedreader_rewriter_service.cc#L36